src/Atmos/Model/ref_state.jl

```diff
@@ -33,9 +34,12 @@ A hydrostatic state specified by a temperature profile and relative humidity.
 struct HydrostaticState{P,F} <: ReferenceState
   temperatureprofile::P
   relativehumidity::F
+  assume_pde_level_balance::Bool
```

Comment:
Thanks: can you describe what this option does?

Reply:
I added a doc string.

Reply:
I still don't quite understand, and the comment above "assumes that the *reference state* is in hydrostatic balance" also is confusing (since it is by definition in hydrostatic balance)?

Reply:
Are you effectively expressing pressure as a perturbation from reference?

Reply:
Pressure ends up being a perturbation from reference, but at the same time gravity includes only density perturbation. This means that the hydrostatic assumption was used directly at the pde level, and not simply by creating a reference state that satisfies it to some precision. As for the above comment, I don't think that arbitrary `ReferenceState` has to be in hydrostatic balance, only those created using `HydrostaticState`. Unless we make that assumption ? The main point of this comment was to stress that we assume this balance only for the reference state, i.e. the model as a whole is is not hydrostatic.

Reply:
@mwarusz I'm also confused here. The reference state is exactly hydrostatic, by construction, so I do not see why we need `assume_pde_level_balance` in `ref_state.jl`. It seems that what you are doing when `assume_pde_level_balance` is true is to factor out the hydrostatic reference state from the model state in the momentum equation, and then work in the perturbation state. You can do that, but it should be an option at the level of the model, not of the reference state.